symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java

```diff
@@ -0,0 +1,520 @@
+
+  /**
+   * Get feature entitlements of an user.
+   *
+   * @param uid User Id
+   * @return    List of feature entitlements of the user.
+   * @see       <a href="https://developers.symphony.com/restapi/reference#features">User Features</a>
+   */
+  public List<Feature> getFeatureEntitlementsOfUser(@NonNull Long uid) {
+    try {
+      return userApi.v1AdminUserUidFeaturesGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Update feature entitlements of an user.
+   *
+   * @param uid       User Id
+   * @param features  List of feature entitlements to be updated
+   * @see             <a href="https://developers.symphony.com/restapi/reference#update-features">Update User Features</a>
+   */
+  public void updateFeatureEntitlementsOfUser(@NonNull Long uid, @NonNull List<Feature> features) {
+    try {
+      userApi.v1AdminUserUidFeaturesUpdatePost(authSession.getSessionToken(), uid, features);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get status of an user.
+   *
+   * @param uid User Id
+   * @return    Status of the user.
+   * @see       <a href="https://developers.symphony.com/restapi/reference#user-status">User Status</a>
+   */
+  public UserStatus getStatusOfUser(@NonNull Long uid) {
+    try {
+      return userApi.v1AdminUserUidStatusGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Update the status of an user
+   *
+   * @param uid     User Id
+   * @param status  Status to be updated to the user
+   * @see           <a href="https://developers.symphony.com/restapi/reference#update-user-status">Update User Status</a>
+   */
+  public void updateStatusOfUser(@NonNull Long uid, @NonNull UserStatus status) {
+    try {
+      userApi.v1AdminUserUidStatusUpdatePost(authSession.getSessionToken(), uid, status);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get user by id
+   *
+   * @param uid   User Id
+   * @param local If true then a local DB search will be performed and only local pod users will be
+   *              returned. If absent or false then a directory search will be performed and users
+   *              from other pods who are visible to the calling user will also be returned.
+   * @return      User found by uid
+   */
+  public UserV2 getUserById(@NonNull Long uid, @NonNull Boolean local) {
+    try {
+      return usersApi.v2UserGet(authSession.getSessionToken(), uid, null, null, local);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get user by id
+   *
+   * @param uid User Id
+   * @return    User found by uid
+   */
+  public UserV2 getUserById(@NonNull Long uid) {
```

Comment:
We could have a getUser(UserId id) method to cover both cases (long id and string email)

And then UserId.id(123) and UserId.email("bob@mail.com")
This way we don't mess up with primitive types and avoid having methods suffixed with byId/byEmail

But maybe the API does not always support id and email?

Reply:
the API for getting user accept only one of the params: username, uid or email. If we pass more than one param, the error 400 is returned. That's why I splitted it to 3 different methods.

Reply:
It does not prevent the UserService to have the logic to set the username/uid/email based on the UserId passed as a parameter

but do **all** the APIs related to user support passing username, uid or email? (otherwise we would not have a consistent API where UserId can be used everywhere)

Reply:
Nope, only the API for getting user: 
https://github.com/symphonyoss/symphony-api-spec/blob/master/pod/pod-api-public.yaml#L1784
and the API for seaching users:
https://github.com/symphonyoss/symphony-api-spec/blob/master/pod/pod-api-public.yaml#L1426